modules/git/blame.go

```diff
@@ -182,33 +191,32 @@ func CreateBlameReader(ctx context.Context, objectFormat ObjectFormat, repoPath
 	}, nil
 }
 
-func tryCreateBlameIgnoreRevsFile(commit *Commit) (string, func()) {
+func tryCreateBlameIgnoreRevsFile(commit *Commit) (string, func(), error) {
 	entry, err := commit.GetTreeEntryByPath(".git-blame-ignore-revs")
 	if err != nil {
-		log.Error("Unable to get .git-blame-ignore-revs file: GetTreeEntryByPath: %v", err)
-		return "", nil
+		if IsErrNotExist(err) {
+			return "", nil, nil
+		}
+		return "", nil, err
```

Comment:
It breaks the design of Golang&Gitea error system

In most cases:

* Return "err" if error occurs
* Return err=nil if no error occurs and then the returned value can't be nil.

But your change breaks this constraint.

----

To make the error handling consistent, do not call `IsErrNotExist` here, let caller check, just like most XORM methods do.

Reply:
3dab8276e2